bundles/net.tourbook/src/net/tourbook/ui/tourChart/SlideoutTourChartGraphs.java

```diff
@@ -29,350 +45,633 @@
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.ToolBar;
 
-import net.tourbook.Messages;
-import net.tourbook.application.TourbookPlugin;
-import net.tourbook.common.action.ActionOpenPrefDialog;
-import net.tourbook.common.font.MTFont;
-import net.tourbook.common.tooltip.ToolbarSlideout;
-import net.tourbook.common.util.Util;
-import net.tourbook.preferences.PrefPageAppearanceTourChart;
-import net.tourbook.tour.TourManager;
-
 /**
  * Tour chart properties slideout.
  */
 public class SlideoutTourChartGraphs extends ToolbarSlideout {
```

Comment:
> The UI looks now OK but the performance to open the slideout is still poor
> 
> Have you removed the nth toolbar resize updates which I commented in a previous review?
> 
> ![mt chart graph slideout](https://user-images.githubusercontent.com/1283445/106466098-2cd81800-649b-11eb-9fbd-317bf33f3652.png)

the last and only review that talked about toolbar performance is this one: https://github.com/wolfgang-ch/mytourbook/pull/296#discussion_r565285032
is this the one you are talking about with "> Have you removed the nth toolbar resize updates which I commented in a previous review?" ?
because I don't understand what you mean by: "> Have you removed the nth toolbar resize updates which I commented in a previous review?"

Reply:
It was in this https://github.com/wolfgang-ch/mytourbook/pull/296#discussion_r565279503 where the `ToolBarManager.relayout()` is called 52 times and needs 961 ms

Reply:
> It was in this [#296 (comment)](https://github.com/wolfgang-ch/mytourbook/pull/296#discussion_r565279503) where the `ToolBarManager.relayout()` is called 52 times and needs 961 ms

So to be clear you want me to change in the function below, 'tbm.update(**true**);' to 'tbm.update(**false**);', correct ?:
private void createUI_GraphAction(final Composite parent, final int graphId) {
      final ToolBar toolbar = new ToolBar(parent, SWT.FLAT);
      final ToolBarManager tbm = new ToolBarManager(toolbar);
      tbm.add(_tourChart.getGraphAction(graphId));
      tbm.update(true);
   }

If it is correct only for the case of custom tracks or for all cases ?

Reply:
> 
> 
> > It was in this [#296 (comment)](https://github.com/wolfgang-ch/mytourbook/pull/296#discussion_r565279503) where the `ToolBarManager.relayout()` is called 52 times and needs 961 ms
> 
> @wolfgang-ch
> So to be clear you want me to change in the function below(which is the one in your code which is called to create the ToolBarManager's), 'tbm.update(**true**);' to 'tbm.update(**false**);', correct ?:

No, you have to replace the 52 toolbars which contain just one action with image buttons for all (old+new) graphs. 

There were much less `tbm.update()` in the old code which was not be recognizable slow, but with 52 `tbm.update()` and 1 sec delay, there must be a faster solution.

You can use image buttons, e.g.

![mt image buttons](https://user-images.githubusercontent.com/1283445/106710568-6d987400-65f6-11eb-9d37-075abea1ba66.png)

Reply:
here you can see where it was already used

![mt image buttons 2](https://user-images.githubusercontent.com/1283445/106710990-21016880-65f7-11eb-873d-66c181b7d9fd.png)

Reply:
> > > It was in this [#296 (comment)](https://github.com/wolfgang-ch/mytourbook/pull/296#discussion_r565279503) where the `ToolBarManager.relayout()` is called 52 times and needs 961 ms
> > 
> > 
> > @wolfgang-ch
> > So to be clear you want me to change in the function below(which is the one in your code which is called to create the ToolBarManager's), 'tbm.update(**true**);' to 'tbm.update(**false**);', correct ?:
> 
> No, you have to replace the 52 toolbars which contain just one action with image buttons for all (old+new) graphs.
> 
> There were much less `tbm.update()` in the old code which was not be recognizable slow, but with 52 `tbm.update()` and 1 sec delay, there must be a faster solution.
> 
> You can use image buttons, e.g.
> 
> ![mt image buttons](https://user-images.githubusercontent.com/1283445/106710568-6d987400-65f6-11eb-9d37-075abea1ba66.png)

> There were much less `tbm.update()` in the old code which was not be recognizable slow, but with 52 `tbm.update()` and 1 sec delay, there must be a faster solution.
ok, I'll explore that but keep in mind that the example I gave is extreme in number of custom tracks, the average guy would have around 10-15 custom tracks not 34 like me.

Reply:
@wolfgang-ch 
here below is my first attempt, is it what you were thinking ? (I merge the text inside the button)
![chart_graph_new](https://user-images.githubusercontent.com/50674761/106825270-d29aab00-6684-11eb-849f-4fbf1bdff253.jpg)

Reply:
This looks like tower doorbells https://www.google.com/search?sa=X&source=univ&tbm=isch&q=%22hochhaus%22+t%C3%BCrklingel&client=firefox-b-d&ved=2ahUKEwj02LWnv9DuAhXHAWMBHVCSDbgQ7Al6BAgFECQ&biw=1498&bih=1375

Reply:
I think it should lool similar like the graph min/max values, 

* Button only with graph image
* Checkbox with graph label as text

Now you can see why I've used a toolbar for each graph (no border, no backgroud color), the UI looks much cleaner but when performance is so bad, then the UI is less important.

![mt graph min max values](https://user-images.githubusercontent.com/1283445/106911852-e5ec5b80-6702-11eb-91c8-27cf05fa5e72.png)

Reply:
> I think it should lool similar like the graph min/max values,
> 
> * Button only with graph image
> * Checkbox with graph label as text
> 
> Now you can see why I've used a toolbar for each graph (no border, no backgroud color), the UI looks much cleaner but when performance is so bad, then the UI is less important.
> 
> ![mt graph min max values](https://user-images.githubusercontent.com/1283445/106911852-e5ec5b80-6702-11eb-91c8-27cf05fa5e72.png)

> I think it should lool similar like the graph min/max values,
> 
> * Button only with graph image
> * Checkbox with graph label as text
@wolfgang-ch 
so like this then ?
![chart_graph_new3](https://user-images.githubusercontent.com/50674761/106960821-f0c3e200-673c-11eb-88a1-12a28c3aff99.jpg)

Reply:
>  so like this then ?
> ![chart_graph_new3](https://user-images.githubusercontent.com/50674761/106960821-f0c3e200-673c-11eb-88a1-12a28c3aff99.jpg)

Yes, how is the performance of this UI, is it faster? Just the feeling if you havn't a profiler.

Reply:
> > so like this then ?
> > ![chart_graph_new3](https://user-images.githubusercontent.com/50674761/106960821-f0c3e200-673c-11eb-88a1-12a28c3aff99.jpg)
> 
> Yes, how is the performance of this UI, is it faster? Just the feeling if you havn't a profiler.

@wolfgang-ch 
I don't have the profiler, but my feeling is that it is faster to display.